composable-tasks/src/compose_tasks.rs

```diff
@@ -0,0 +1,137 @@
+                        "EGLD can't be swapped!"
+                    );
+                    let payment_in = payment_for_current_task.unwrap_esdt();
+
+                    require!(args.len() == 2, "Swap requires only 2 arguments");
+
+                    let token_out = TokenIdentifier::from(args.get(0).clone_value());
+                    let min_amount_out = BigUint::from(args.get(1).clone_value());
+
+                    self.perform_tokens_swap(
+                        payment_in.token_identifier,
+                        payment_in.amount,
+                        token_out,
+                        min_amount_out,
+                    )
+                    .into()
+                }
+                TaskType::RouterSwap => {
+                    require!(
+                        !payment_for_current_task.token_identifier.is_egld(),
+                        "EGLD can't be swapped!"
+                    );
+                    require!(
+                        args.len() % 4 == 0,
+                        "Invalid number of router swap arguments"
+                    );
+                    let payment_in = payment_for_current_task.unwrap_esdt();
+                    let mut returned_payments_by_router = self.multi_pair_swap(payment_in, args);
+
+                    let last_payment_index = returned_payments_by_router.len() - 1;
+                    let payment_out = returned_payments_by_router.take(last_payment_index);
+                    payments_to_return.append_vec(returned_payments_by_router);
+                    EgldOrEsdtTokenPayment::from(payment_out)
+                }
+                TaskType::SendEgldOrEsdt => {
+                    self.require_min_expected_token(
+                        &min_expected_token_out,
+                        &payment_for_current_task,
+                    );
+                    require!(args.len() == 1, "Invalid number of arguments!");
+
+                    let dest_addr =
+                        ManagedAddress::try_from(args.get(0).clone_value()).unwrap_or(caller);
+
+                    if payment_for_current_task.token_identifier.is_egld() {
+                        self.send()
+                            .direct_egld(&dest_addr, &payment_for_current_task.amount);
+                    } else {
+                        payments_to_return.push(EsdtTokenPayment::new(
+                            payment_for_current_task.token_identifier.unwrap_esdt(),
+                            payment_for_current_task.token_nonce,
+                            payment_for_current_task.amount,
+                        ));
+                    }
+                    if !payments_to_return.is_empty() {
+                        self.send().direct_multi(&dest_addr, &payments_to_return);
+                    }
+                    return;
+                }
+            };
+        }
+        self.require_min_expected_token(&min_expected_token_out, &payment_for_next_task);
```

Comment:
This part is the same as the end of the function. Make a general purpose function for sending payments and send the corresponding arguments to it, based on the context.